src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzerTests.cs

```diff
@@ -0,0 +1,1045 @@
+    {
+    }
+}
+" + MockAttributesCsSource;
+            await VerifyAnalyzerAsyncCs(source);
+        }
+
+        [Fact]
+        public async Task LambdaCallsOsDependentMemberWarns()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+using System;
+
+public class Test
+{
+    public void M1()
+    {
+        void Test() => [|M2()|];
+        Test();
+
+        Action action = () =>
+        {
+            [|M2()|];
+        };
+    }
+
+    [MinimumOSPlatform(""Windows10.1.2.3"")]
+    public void M2()
+    {
+    }
+}
+" + MockAttributesCsSource;
+            await VerifyAnalyzerAsyncCs(source);
+        }
+
+        [Fact]
+        public async Task AttributedLambdaCallsOsDependentMemberNotWarn()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+using System;
+
+public class Test
+{
+    [MinimumOSPlatform(""Windows10.13"")]
+    public void M1()
+    {
+        void Test() => M2();
+        Test();
+
+        Action action = () =>
+        {
+            M2();
+        };
+    }
+
+    [MinimumOSPlatform(""Windows10.1.2.3"")]
+    public void M2()
+    {
+    }
+}
+" + MockAttributesCsSource;
+            await VerifyAnalyzerAsyncCs(source);
+        }
+
+        [Fact]
+        public async Task OsDependentEventAccessedWarns()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+public class Test
+{
+    public delegate void Del();
+
+    [MinimumOSPlatform(""Windows10.1.2.3"")]
+    public event Del SampleEvent;
+
+    public void M1()
+    {
+        [|SampleEvent|] += M3;
+        M2();
+    }
+
+    public void M2()
+    {
+        [|SampleEvent|]?.Invoke();
+    }
+
+    public void M3()
+    {
+    }
+}
+" + MockAttributesCsSource;
+            await VerifyAnalyzerAsyncCs(source);
+
+            var vbSource = @"
+Imports System.Runtime.Versioning
+
+Public Class Test
+    Public Delegate Sub Del()
+    <MinimumOSPlatform(""Windows10.1.2.3"")>
+    Public Event SampleEvent As Del
+
+    Public Sub M1()
+        AddHandler [|SampleEvent|], AddressOf M3
+        M2()
+    End Sub
+
+    Public Sub M2()
+        RaiseEvent  [|SampleEvent|]
+    End Sub
+
+    Public Sub M3()
+    End Sub
+End Class"
++ MockAttributesVbSource;
+            await VerifyAnalyzerAsyncVb(vbSource);
+        }
+
+        [Fact]
+        public async Task OsDependentMethodAssignedToDelegateWarns()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+public class Test
+{
+    public delegate void Del(); // The attribute not supported on delegates, so no tets for that
+
+    [MinimumOSPlatform(""Windows10.1.2.3"")]
+    public void DelegateMethod()
+    {
+    }
+    public void M1()
+    {
+        Del handler = [|DelegateMethod|];
+        handler(); // assume it shouldn't warn here
+    }
+}
+" + MockAttributesCsSource;
+            await VerifyAnalyzerAsyncCs(source);
+        }
+
+        /*[Fact] TODO wait until assembly level APIs merged
```

Comment:
Why this TODO if the analyzer is anyways working off mock attribute sources?

Reply:
i was not able to add 2 assemblies in one source, so was not able to test assembly-level attributes, yes you told me that i can use 2 sources for testing but i haven't figured how to do that yet and didn't found any example ...

Reply:
I think there is some confusion around assembly level attributes. Isn't it possible to add the below test without use of multiple source files or assemblies?

```csharp
[assembly: MinimumOSPlatform("Windows10")]

class C
{
    void Windows10Method()
    {
        Windows11Method(); // This should be flagged?
    }

    [MinimumOSPlatform("Windows11")]
    void Windows11Method()
    {
    }
}
```

Reply:
Yeah, i could have added some scenario like this, but it couldn't cover all scenarios, for this it is same as awarning for a method level attribute

Reply:
We now have some tests for assembly-level attributes, but still can't test the scenario between different assemblies, this TODO is a reminder for adding them after tests can consume attributed APIs